### ISSUE TITLE
Multi-document publications and meaningful sequence

### ISSUE THREAD

**Issue by mattgarrish** [w3c/wcag21#162]
The Understanding WCAG document addresses meaningful sequence within a single page for SC 1.3.2, but does not address the importance of the logical reading order when a document is split across multiple pages. If the sequence is not maintained across page boundaries and there isn't a mechanism to continue reading in sequence, a publication won't be readable.

This issue popped up in a recent discussion in the DPUB IG, where there was some confusion about whether WCAG has any requirements for reading order across documents, so we'd like to propose a couple of changes to the Understanding document to better highlight the applicability.

First, we propose adding the following new paragraph to the Intent section:

> If content is split across multiple pages, the reading order also needs to be preserved across page boundaries, and a means of following the sequence from page to page provided. For example, if each page contains one chapter of a publication, it must be possible to follow from the end of one chapter to the start of the next.

This paragraph could be added after the one that begins "A sequence is meaningful if the order of content cannot be changed ...".

Second, we suggest adding an additional success example:

> Example 3: In a publication split across multiple pages, the reading order is preserved across page boundaries such that the end of one document flows logically into the top of the next. A mechanism is available that allows the user, or user agent, to move from document to document in their logical sequence.

**Comment by clapierre** [comment]
I support this and also would like to point out that this was gap pointed out last year in the Digital Publishing's Interest Groups [Accessibility Note](https://www.w3.org/TR/dpub-accessibility/#navigating-within-multiple-documents-or-packaged-documents).

**Comment by awkawk** [comment]
It is accurate that WCAG's entire conformance model is based on the "page" and I thought that our earlier discussion was around a reading order for an EPUB document. The EPUB document is a "web page" in that it is "a non-embedded resource obtained from a single URI using HTTP plus any other resources that are used in the rendering or intended to be rendered together with it by a user agent", so 1.3.2 applies to a multi-page EPUB document in the same way that it applies to a multi-page PDF document.

Is this the issue that you are trying to address understanding of the applicability of 1.3.2 to, or is it something else?

If a long publication was split into different files that were needed to access the publication (e.g. a separate EPUB file for each volume of an encyclopedia, for some reason) 1.3.2 would only apply to each EPUB document on its own. I agree that if there wasn't a way to continue reading a document after volume 1 to get to volume 2 that would be a problem, but think that in the context of WCAG 2.0/2.1 that we would view that as a general usability problem rather than a specific accessibility concern.

We should add to the Understanding document (http://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-sequence.html):
At the end of the Intent section, add 
"Meaningful sequence applies to technologies that support paginated views of content such as EPUB and PDF. Each document which is able to be obtained from a single URI is considered a Web page by WCAG 2.0 and the author needs to ensure that the content is presented in a meaningful sequence."
And then perhaps an example in the examples section?

**Comment by mattgarrish** [comment]
This was raised around the work being chartered on web publications, which would see publications hosted unpackaged on the web. EPUB would probably benefit from the clarification, though.

The question is sequence in any multi-page document. It could be a book that is fractured across pages, but even articles you find on the web now are typically broken across many pages to increase the number of ads you get served. Sequence within a single page is important, but a document is not accessible if you cannot logically read it in its entirety across the full set of pages.

**Comment by DavidMacDonald** [comment]
Although I wish meaningful sequence applied across pages, WCAG is page based. I think that it would have to have the phrase "set of pages" in it, in order to apply across pages. I think we should amend it in 2.1. 

1.3.2 Meaningful Sequence: When the sequence in which content is presented `<add>`on the page or across a set of pages `</add>`affects its meaning, a correct reading sequence can be programmatically determined. (Level A)


The only other part of the standard that I think could be involked conformance requirement 2:

>3. Complete processes: When a Web page is one of a series of Web pages presenting a process (i.e., a sequence of steps that need to be completed in order to accomplish an activity), all Web pages in the process conform at the specified level or better. (Conformance is not possible at a particular level if any page in the process does not conform at that level or better.)

**Comment by awkawk** [comment]
Are there examples where this is seen as a problem?

As I think about this more I'm wondering how much of this represents issues for all users ("ok, I just read page 1 and now it is sending me to page 5 and I can't find a way to get to page 2" - this would affect all users equally) and how much represents a need for multipage content to be diligent in applying 1.3.2 to each page.

@clapierre I can't find where this is really discussed in the note you linked to. what section?

**Comment by mattgarrish** [comment]
A real-world example from epub is publishers who don't put content documents in the spine, or put them in the wrong order. It's not intentional, of course, but happens through lack of diligence when editing and updating content.

Given that the web publications specification will (as far as I'm aware at this time) have a similar out-of-band mechanism for the reading order of documents, the problem is going to surface with it, too.

I looked at multiple ways in hopes that it already addressed sequence, but it only requires some way to reach the documents, not that there be a logical progression through them.

But, granted, an inability to sequentially move from document to document will affect all users. Is that a disqualifier?

I don't want to overemphasize the mechanism for linking, though, as it's not important how that is done. If we can highlight that sequence is also important across a set of pages, then by default broken linking, however it manifests, would be a failure.

A change like @DavidMacDonald suggested looks fine to me if you want to drop the prose addition to the understanding document. If we could keep the example and strip the last sentence, I think we've accomplished enough for publications.

**Comment by mattgarrish** [comment]
Another quick thought, but with publications, non-essential documents also get interleaved with essential (the concept of linear and non-linear documents in epub). It's probably too complicated for 2.1 and before web publications becomes a reality, but it's another sequencing problem in that you don't want the non-linear documents placed where they will interrupt the primary narrative (e.g., they should occur after major sections or at the end of the publication, not interspersed where they can break the reading order.

**Comment by awkawk** [comment]
@mattgarrish I'm not sure what "publishers who don't put content documents in the spine" means in a technical sense.

Regarding whether not being able to move from document to document affecting all users being a disqualifier, I may be misunderstanding the situation, but why would anyone publish a book like that? If you can't get to the whole book that's a problem for sure, but it doesn't feel like an accessibility problem so much as a bug in the book type of problem.

**Comment by mattgarrish** [comment]
The spine defines the default sequence of content documents in epub. It's external metadata from the content documents themselves (i.e., you don't click a link to move from one document to the next; the set is seamless to the end user). Web publications will have a similar external ordering mechanism, if not exactly the same as the epub spine (that's still up for debate). 

The linking isn't so much the key, though. You just can't have a meaningful sequence across a set of pages if there isn't also a meaningful progression to get from one to the next. When that progression is broken, the publication is junk. It does happen, if not commonly, as automated validation can't alert you whether documents are out of sequence.

The concern is more that publishers keep in mind that sequence doesn't necessarily end with one document and begin completely anew with the next when the documents are logically all a set. The publication in total has a meaningful sequence, and the author needs to consider the flow from one document to the next, as well as within each document.

That's why I'm personally fine with not getting into the ordering mechanism, as if it's clear that sequence has to be meaningful across the set then there has to be a logical progression to maintain the sequence.

**Comment by DavidMacDonald** [comment]
@mattgarrish Are you looking for something like this?

> 1.3.2 Meaningful Sequence: When the sequence in which content is presented affects its meaning,  when sequentially related content is divided across  a set of pages, a correct reading sequence can be programmatically determined,  (Level A)

>sequentially related content: Definition here.

**Comment by eadraffan** [comment]
Looks good, but does this mean only content where the meaning is dependent on the previous section, paragraph or sentence?

 

 

Best wishes

E.A. 

 

Mrs E.A. Draffan

WAIS, ECS , University of Southampton

Mobile +44 (0)7976 289103

http://access.ecs.soton.ac.uk <https://www.outlook.soton.ac.uk/owa/redir.aspx?C=69b1RzNTDwem3wbm4pLRmuYfTLt16YjcghtEpZBsF5Sebx78I2DUCA..&URL=http%3a%2f%2faccess.ecs.soton.ac.uk%2f> 

UK AAATE rep http://www.aaate.net/ <https://www.outlook.soton.ac.uk/owa/redir.aspx?C=WUwOCw_4FszLSzcUbkoFdDkad8-Q_GrRfPYUJ_ol5l2ebx78I2DUCA..&URL=http%3a%2f%2fwww.aaate.net%2f>

**Comment by mattgarrish** [comment]
Is it a significant enough distinction to warrant a separate success criteria? We'd be pulling apart page from set of pages but keeping the underlying need.

But I tend to only focus on publications, so if you think there's broader applicability that would be served by splitting off the set concept, then I defer to your judgement.

I think we'd be okay with small changes, like you had proposed earlier just to note a set of pages in the SC, or only adding to the understanding document.

/cc @clapierre @georgekerscher @avneeshsingh

**Comment by awkawk** [comment]
(Official WG response)
Thank you for the comment. The Working Group will add content to the Understanding document as indicated in the issue responses. We will also mark this issue as "defer" to ensure that it is reviewed for more comprehensive changes in the Silver timeframe.
